File: Sorting/selection sort algorithm/selection_sort.py
def selection_sort(arr):
    n = len(arr)

    for i in range(n - 1):
        # Find the minimum element in the unsorted portion
        min_index = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_index]:
                min_index = j

        # Swap the minimum element with the first element of the unsorted portion
        arr[i], arr[min_index] = arr[min_index], arr[i]


# Example usage
array = [64, 25, 12, 22, 11]
selection_sort(array)
print("Sorted array:", array)

File: Sorting/selection sort algorithm/selection_sort.js
function selectionSort(arr) {
  const n = arr.length;

  for (let i = 0; i < n - 1; i++) {
    let minIndex = i;

    // Find the index of the minimum element in the unsorted portion
    for (let j = i + 1; j < n; j++) {
      if (arr[j] < arr[minIndex]) {
        minIndex = j;
      }
    }

    // Swap the minimum element with the first element of the unsorted portion
    [arr[i], arr[minIndex]] = [arr[minIndex], arr[i]];
  }
}

// Example usage
let array = [64, 25, 12, 22, 11];
selectionSort(array);
console.log("Sorted array:", array);
